pkg/operator/controller/externaldns/controller.go

```diff
@@ -110,10 +110,6 @@ func (r *reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
 		return reconcile.Result{}, fmt.Errorf("failed to ensure externalDNS cluster role: %w", err)
 	}
 
-	if _, _, err := r.ensureExternalDNSNamespace(ctx, r.config.Namespace); err != nil {
-		return reconcile.Result{}, fmt.Errorf("failed to ensure externalDNS namespace: %w", err)
-	}
```

Comment:
I'm not sure about this one. It still makes sense for non OLM deployments.

We can do:
- Miciah proposed: just keep it as-is 
OR
- don't ensure the namespace in case the platform is OpenShift

Reply:
We can remove `ensureExternalDNSNamespace` for now if @dbaker-rh prefers we do so.  The code is still there in Git history for such a time as we get the RBAC issues sorted out.